docs/mine/lotus/message-pool.md

```diff
@@ -0,0 +1,76 @@
+---
+title: 'Lotus Miner: the message pool'
+description: 'The Message Pool (mpool) is the component of lotus that handles pending messages for inclusion in the chain.'
+breadcrumb: 'Message Pool'
+---
+
+# {{ $frontmatter.title }}
+
+{{ $frontmatter.description }} Messages are added to the mpool either directly for locally published messages or through pubsub propagation. Whenever a miner is ready to create a block for a tipset, it invokes the mpool selection algorithm which selects an appropriate set of messages such that it optimizes miner reward and chain capacity.
+
+When messages are executed, they use _gas_. The amount of _gas_ used, the parameters attached to each message and the Network's current _BaseFee_ determine how much FIL will be spent to include that message in the chain and how much of that expense will be given to the miner as reward, or burned.
+
+The different caps and fees for messages are explained below, along with instructions on how to inspect and modify the current local message pool.
+
+::: tip
+Lotus provides the tooling to interact with the message pool with the `lotus mpool` subcommands.
+:::
+
+[[TOC]]
+
+## Message selection
+
+When mining a new block, miners must select a set of messages inclusion in a way that maximizes the reward. Given how message distribution works, using pubsub, and that miners do not communicate their tickets to each other, it is not possible to fully determine if a different miner will also include the message on a different block in the new tipset, and perhaps get the reward instead if their block executes first. The problem is NP-hard (a instance of knapsack packing) so, at best, an approximation can be made to the optimal selection in a reasonable amount of time.
+
+Lotus employs a sophisticated algorithm for selecting messages for inclusion from the pool, given the ticket quality of a miner. The ticket quality reflects the probability of execution order for a block in the tipset. Given the ticket quality the algorithm computes the probability of each block, and picks dependent chains of messages such that the reward is maximized, while also optimizing the capacity of the chain.
+
+If the ticket quality is sufficiently high, then a greedy selection algorithm is used instead that simply picks dependent chains of maximum reward. Note that pending message chains from priority addresses are always selected, regardless of their profitability.
+
+## Gas, fees, limits and caps
```

Comment:
This would be a good place to include a reference to setting up a control address so that wpost messages are prioritized.

Reply:
Added tip